src/savegame_json.cpp

```diff
@@ -4999,10 +5009,22 @@ void submap::load( const JsonValue &jv, const std::string &member_name, int vers
                     if( !remaining ) {
                         JsonValue terrain_entry = terrain_json.next_value();
                         if( terrain_entry.test_string() ) {
-                            iid = ter_str_id( terrain_entry.get_string() ).id();
+                            const ter_str_id terstr( terrain_entry.get_string() );
```

Comment:
I'd like if we could have some sort of JSON-defined migration for this, but I understand that's even more scope for this PR.

Reply:
I'd also appreciate if you could add a test for this (tests/submap_load_test.cpp), but it's by no means required.

Reply:
I added the test just in case, but I think migrating terrain ids is best left for a separate PR

Reply:
I agree. Thank you!